tensorboard/backend/event_processing/data_ingester_test.py

```diff
@@ -252,45 +255,71 @@ def testSingleLetterGroup(self):
             )
 
 
-class FileSystemSupport(tb_test.TestCase):
+class ImportMock(object):
```

Comment:
This approach is fine, but FWIW, it shouldn't really be necessary to create a persistent object (and store it on the test class) for this, since in Python it's quite easy to make nested functions and closures.  So you could consider an approach like this which creates a general utility:

```python
_ORIGINAL_IMPORT = __import__

def fake_import(affected_name, *, success=True):
    def fake(name, *args, **kwargs):
        if name == affected_name:
            if success:
                return
            else:
                raise ImportError("Pretending I'm missing ;)")
        return _ORIGINAL_IMPORT(name, *args, **kwargs)
    return fake
```

Then in your code you could use it like this:

```python
with mock.patch(
    "builtins.__import__",
    side_effect=fake_import("tensorflow_io", success=True),
) as mock_import:
    # do stuff
```

Reply:
Done. Thanks!